internal/controller/documentdb_controller.go

```diff
@@ -75,7 +76,8 @@ func (r *DocumentDBReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 
 		// Ensure LoadBalancer has an IP assigned
-		if err := util.EnsureLoadBalancerIP(ctx, foundService); err != nil {
+		loadBalancerIp, err = util.EnsureLoadBalancerIP(ctx, foundService)
+		if err != nil {
 			log.Info("DocumentDB LoadBalancer External IP not assigned, Requeuing.")
```

Comment:
does this only work for external IP? Ehat about internal?

Reply:
Now supporting both public loadbalancer and Cluster IP service.